File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-1_Gestion-Salaires-Employes/Implementation-Cpp/Commercial.cpp
#include "Commercial.h"

const double Commercial::BASE = 2000.0;
const double Commercial::PART = 0.2;

Commercial::Commercial(string nom, int age, int anciennete, double chiffreAffaire) : Employe(nom, age, anciennete) {
    this->chiffreAffaire = chiffreAffaire;
}

void Commercial::afficher() {
    cout << "Commercial : " << nom;
}

double Commercial::calculerBaseSalaire() {
    return BASE + (chiffreAffaire * PART);
}

File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-1_Gestion-Salaires-Employes/Implementation-Java/Commercial.java
class Commercial extends Employe {
    private double chiffreAffaire;
    private static final double BASE = 2000.0;
    private static final double PART = 0.2;

    public Commercial(String nom, int age, int anciennete, double chiffreAffaire) {
        super(nom, age, anciennete);
        this.chiffreAffaire = chiffreAffaire;
    }

    @Override
    public void afficher() {
        System.out.print("Commercial : " + nom);
    }

    @Override
    public double calculerBaseSalaire() {
        return BASE + (chiffreAffaire * PART);
    }
}
